genestack_client/scripts/genestack_user_setup.py

```diff
@@ -38,6 +38,52 @@ def ask_alias(existed):
         return alias
 
 
+def ask_use_token():
+    items = ['by token', 'by email and password']
+
+    while True:
+        for i, option in enumerate(items, start=1):
+            print '%s) %s' % (i, option)
+
+        raw_index = raw_input("Select authentication: ").strip()
+
+        try:
+            user_index = int(raw_index) - 1
+        except ValueError:
+            print 'Wrong number: "%s".' % raw_index
+            continue
+
+        try:
+            return items[user_index]
+        except IndexError:
+            print "Number is not in list."
+            continue
+
+
+def ask_auth(host, alias=None):
+    if ask_use_token():
```

Comment:
~`ask_use_token`~`select_auth_method` returns non-empty string, not a boolean value.

Reply:
Also I don't think this function needs to be separate from ~`ask_use_token`~`select_auth_method`.  Merge them together into `select_auth_method(host, alias=None)`.